src/GifWin/App.xaml.cs

```diff
@@ -189,13 +189,15 @@ void ConvertGifWitLibrary ()
                 Multiselect = false,
             };
 
-            var result = dlg.ShowDialog ();
+            var result = Dispatcher.Invoke(() => dlg.ShowDialog ());
```

Comment:
Whoops, forgot to remove the dispatching bits in this method when I rejiggered this. Good catch.